File: CS-Examples/04_RowsColumns/InsertRowsAndColumns.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace InsertRowsAndColumns
{
	public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }		
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a new workbook object
            Workbook workbook = new Workbook();

            // Load an existing file from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\InsertRowsAndColumns.xls");

            // Get the first worksheet in the workbook
            Worksheet worksheet = workbook.Worksheets[0];

            // Insert a row into the worksheet at index 2
            worksheet.InsertRow(2);

            // Insert a column into the worksheet at index 2
            worksheet.InsertColumn(2);

            // Insert multiple rows into the worksheet starting at index 5, with a count of 2
            worksheet.InsertRow(5, 2);

            // Insert multiple columns into the worksheet starting at index 5, with a count of 2
            worksheet.InsertColumn(5, 2);

            // Specify the output file name
            string result = "InsertRowsAndColumns_out.xlsx";

            // Save the modified workbook to the specified file using Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // View the file
            ExcelDocViewer(result);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: VB-Examples/04_RowsColumns/InsertRowsAndColumns.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace InsertRowsAndColumns
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Creates a new instance of the Workbook class.
            Dim workbook As New Workbook()
            'Loads the Excel document from the specified path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\InsertRowsAndColumns.xls")
            'Retrieves the first worksheet from the workbook.
            Dim worksheet As Worksheet = workbook.Worksheets(0)
            'Inserting a row into the worksheet.
            worksheet.InsertRow(2)
            'Inserts the second column into the worksheet.
            worksheet.InsertColumn(2)
            'Inserts two rows from the fifth row into the worksheet.
            worksheet.InsertRow(5, 2)
            'Inserting two columns from the fifth column into the worksheet.
            worksheet.InsertColumn(5, 2)
            'Specifies the name of the resulting Excel file.
            Dim result As String = "InsertRowsAndColumns_out.xlsx"
            'Saves the modified workbook to a file with the specified name and Excel version.
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer(result)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
